unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -0,0 +1,310 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCl.Social.Friends;
+using DCL.Social.Friends;
+using Decentraland.Renderer.RendererServices;
+using Decentraland.Social.Friendships;
+using Google.Protobuf.WellKnownTypes;
+using JetBrains.Annotations;
+using RPC;
+using System;
+using System.Threading;
+using rpc_csharp;
+using rpc_csharp.transport;
+using RPC.Transports;
+using System.Collections.Generic;
+using UnityEngine;
+using Payload = Decentraland.Social.Friendships.Payload;
+
+namespace MainScripts.DCL.Controllers.FriendsController
+{
+    public class RPCSocialApiBridge : IRPCSocialApiBridge
+    {
+        private const int REQUEST_TIMEOUT = 30;
+
+        private string accessToken;
+
+        private readonly IUserProfileBridge userProfileWebInterfaceBridge;
+
+        public event Action<UserStatus> OnFriendAdded;
+        public event Action<string> OnFriendRemoved;
+        public event Action<FriendRequest> OnFriendRequestAdded;
+        public event Action<string> OnFriendRequestRemoved;
+        public event Action<AddFriendRequestsPayload> OnFriendRequestsAdded;
+        public event Action<FriendshipUpdateStatusMessage> OnFriendshipStatusUpdated;
+        public event Action<FriendRequestPayload> OnFriendRequestReceived;
+
+        private ClientFriendshipsService socialClient;
+
+        public RPCSocialApiBridge(MatrixInitializationBridge matrixInitializationBridge, IUserProfileBridge userProfileWebInterfaceBridge)
```

Comment:
We should depend upon an abstraction `IMatrixInitializationBridge`. Otherwise mocking in tests will be painful.

Reply:
Created the interface